core/sector_storage/impl/local_worker.hpp

```diff
@@ -8,63 +8,78 @@
 #include "sector_storage/worker.hpp"
 
 #include "common/logger.hpp"
-#include "sector_storage/stores/impl/local_store.hpp"
 #include "proofs/impl/proof_engine_impl.hpp"
+#include "sector_storage/stores/impl/local_store.hpp"
 
 namespace fc::sector_storage {
 
   struct WorkerConfig {
-    primitives::sector::RegisteredSealProof seal_proof_type;
+    boost::optional<std::string> custom_hostname = boost::none;
     std::set<primitives::TaskType> task_types;
+    bool is_no_swap = false;
   };
 
-  class LocalWorker : public Worker {
+  class LocalWorker : public Worker,
+                      public std::enable_shared_from_this<LocalWorker> {
    public:
-    LocalWorker(WorkerConfig config,
-                std::shared_ptr<stores::RemoteStore> store,
-                std::shared_ptr<proofs::ProofEngine> proofs =
-                        std::make_shared<proofs::ProofEngineImpl>());
+    /**
+     * @note Factory method simply replace constructor, because we need at least
+     * 1 shared_ptr for correct job
+     */
+    static std::shared_ptr<LocalWorker> newLocalWorker(
+        std::shared_ptr<boost::asio::io_context> context,
+        const WorkerConfig &config,
+        std::shared_ptr<WorkerReturn> return_interface,
+        std::shared_ptr<stores::RemoteStore> store,
+        std::shared_ptr<proofs::ProofEngine> proofs =
+            std::make_shared<proofs::ProofEngineImpl>());
+
+    outcome::result<CallId> addPiece(
+        const SectorRef &sector,
+        gsl::span<const UnpaddedPieceSize> piece_sizes,
+        const UnpaddedPieceSize &new_piece_size,
+        PieceData piece_data) override;
 
-    outcome::result<PreCommit1Output> sealPreCommit1(
-        const SectorId &sector,
+    outcome::result<CallId> sealPreCommit1(
+        const SectorRef &sector,
         const SealRandomness &ticket,
         gsl::span<const PieceInfo> pieces) override;
 
-    outcome::result<SectorCids> sealPreCommit2(
-        const SectorId &sector,
+    outcome::result<CallId> sealPreCommit2(
+        const SectorRef &sector,
         const PreCommit1Output &pre_commit_1_output) override;
 
-    outcome::result<Commit1Output> sealCommit1(
-        const SectorId &sector,
-        const SealRandomness &ticket,
-        const InteractiveRandomness &seed,
-        gsl::span<const PieceInfo> pieces,
-        const SectorCids &cids) override;
+    outcome::result<CallId> sealCommit1(const SectorRef &sector,
+                                        const SealRandomness &ticket,
+                                        const InteractiveRandomness &seed,
+                                        gsl::span<const PieceInfo> pieces,
+                                        const SectorCids &cids) override;
 
-    outcome::result<Proof> sealCommit2(
-        const SectorId &sector, const Commit1Output &commit_1_output) override;
+    outcome::result<CallId> sealCommit2(
```

Comment:
Please select more detailed names than sealCommit1 & sealCommit2

Reply:
or add comments

Reply:
This standard name of operation